datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -484,6 +612,12 @@ impl Stream for GroupedHashAggregateStream {
                         (
                             if self.input_done {
                                 ExecutionState::Done
+                            } else if self
```

Comment:
nit is that putting this into a function (like `self.should_skip_aggregation()`) would make this logic easier to follow

Reply:
Filed https://github.com/apache/datafusion/pull/11821# with a proposal for this change